### PR TITLE
Add Sketchfab and Thingiverse as additional sources for 3D models

### LINKED ISSUE #928: Add types to `utils/get-legacy-source-url.js`
## Description

Add type checking to `get-legacy-source-url.js`. This module has one external dependency which already has published types. Make sure to add it to `tsconfig.json`'s `include` list.

## Additional context

Part of an ongoing effort to add type checking to parts of the project that can be type checked.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### LINKED ISSUE #1119: Add Thingiverse and Sketchfab as `model_3d` meta search items
## Problem

We don't have any 3d model meta search sources defined yet. We should add Thingiverse and Sketchfab, since these are our existing 3D sources but we have very few items from them. These sources will appear once #739 is completed.

## Description

Sketchfab and Thingiverse need to be added to the legacy source lookup. The key for these items should be `model_3d`:

https://github.com/WordPress/openverse-frontend/blob/f18e90a92e834cd650fbcf1f535df57918b17a0d/src/utils/get-legacy-source-url.js#L7

- SketchFab URL (license filters seem to be long unique identifiers `https://sketchfab.com/search?features=downloadable&licenses=322a749bcfa841b29dff1e8a1bb74b0b&licenses=7c23a1ba438d4306920229c12afcb5f9&licenses=34b725081a6a4184957efaec2cb84ed3&licenses=2628dbe5140a4e9592126c8df566c0b7&licenses=bbfe3f7dbcdd4122b966b85b9786a989&licenses=72360ff1740d419791934298b8b6d270&licenses=b9ddc40b93e34cdca1fc152f39b9f375&q=dogs&type=models`
- Thingiverse URL `https://www.thingiverse.com/search?q=dog&type=things&sort=relevant` filtering by license doesn't currently appear to work, but the vast majority of items use CC licenses

### LINKED ISSUE #1194: Rename `get-legacy-source-url` to `get-additional-sources`
## Problem

The term 'legacy source url' is unhelpful and is more easily understood as 'get additional sources'. We should update all references to this utility along with the file itself.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #1194 by @zackkrida 
Fixes #1119 by @zackkrida 
Fixes #928 by @sarayourfriend 

<s>Parts of this PR conflict with #1190, which is unfortunate but unavoidable given the nature of the work.</s> Resolved in #1204.

## Description

This PR
- adds a 3D models media type
- renames legacy sources to additional sources
- adds Sketchfab and Thingiverse as sources for 3D models
- converts the util and 4 other components to TypeScript
- refactors to code to move most of the logic from components to the util
- updates tests

### Dev server

0. Check out the PR and run the dev server.
1. Search for a term with no results.
2. See the meta-search links.
3. Change the media type.
4. Observe the meta-search links change.

### Storybook

To see all media types and their search results, use the stories for VMetaSourceList in the Storybook.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
Looks great so far. How does one "Change the media type." to 3d models, as described in the testing instructions?

**Comment by zackkrida** [comment]
Seeing odd behavior that seems almost race-condition-ey. Not sure if it's part of this PR or the result of state management changes elsewhere, I'll test that now.

https://user-images.githubusercontent.com/6351754/160847460-5356d17b-f686-4e23-a0ff-d196e6c75489.mp4

**Comment by zackkrida** [comment]
It's not part of this PR, it's likely the result of the media provider refactor. I'm able to replicate it here: https://search-staging.openverse.engineering/search/audio?q=this%20search%20returns%20no%20results%20amirite as well.

**Comment by dhruvkb** [comment]
Cool, that had me worried because this PR doesn't touch any of that code at all!

File: src/utils/get-additional-sources.ts [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+})
+
+type SearchFunctions = {
+  [k in MediaType]?: (
+    search: AdditionalSearchQuery
+  ) => BuildUrlOptions & { url: string }
+}
+
+/**
+ * , which contains a display name, whether the
```
Comment (obulat) [inline src/utils/get-additional-sources.ts:44]: Something is missing in the docstring

**Review by obulat — APPROVED** [review]
Really nice clean-up of the additional sources utils! I checked the links, too, and they all seem to work, although not many return results for cats.
I should have reviewed this PR before starting on 1233 :) Do the other components here not work without TS in Button and Icon?
I have added more changes to TS-ified `VButton` in #1233. Namely, by extracting the button types to a separate file, we can then import the types to `VIconButton` that uses them, too. Well, I can rebase those changes on top of yours after this PR is merged.

File: src/utils/get-additional-sources.ts (hunk 1 above)
Comment (dhruvkb) [reply src/utils/get-additional-sources.ts:44]: I'm confused as to how that bit got deleted! Thanks for catching this.

**Comment by dhruvkb** [comment]
I had glanced at #1233, it's a much more comprehensive PR, thanks for agreeing to rebase it 😄.

**Review by krysal — APPROVED** [review]
I should have read beyond **Dev server** instructions, then this is good to go! Checked Storybook and looks pretty good!

**Pull request merged**